met-web/src/components/engagement/listing/index.tsx

```diff
@@ -30,88 +30,79 @@ import AdvancedSearch from './AdvancedSearch/SearchComponent';
+    const engagementPage = fetcherData ? fetcherData.engagements : undefined;
+
     const [paginationOptions, setPaginationOptions] = useState<PaginationOptions<Engagement>>({
-        page: Number(pageFromURL) || 1,
-        size: Number(sizeFromURL) || 10,
-        sort_key: 'created_date',
-        nested_sort_key: 'engagement.created_date',
-        sort_order: 'desc',
+        page: Number(searchParams.get('page')) || 1,
+        size: Number(searchParams.get('size')) || 10,
+        sort_key: (searchParams.get('sort_key')?.split('.')?.[1] as keyof Engagement) ?? 'created_date',
+        nested_sort_key: searchParams.get('sort_key') || 'engagement.created_date',
+        sort_order: (searchParams.get('sort_order') as 'asc' | 'desc') ?? 'desc',
     });
 
     const [pageInfo] = useState<PageInfo>(createDefaultPageInfo());
 
+    const [searchText, setSearchText] = useState(searchParams.get('search_text') || '');
     const [advancedSearchOpen, setAdvancedSearchOpen] = useState(false);
 
     const [searchOptions, setSearchOptions] = useState<SearchOptions>({
-        status_list: [],
-        created_from_date: '',
-        created_to_date: '',
-        published_from_date: '',
-        published_to_date: '',
+        status_list: (searchParams.getAll('engagement_status') || []).map((status) => Number(status)),
+        created_from_date: searchParams.get('created_from_date') || '',
+        created_to_date: searchParams.get('created_to_date') || '',
+        published_from_date: searchParams.get('published_from_date') || '',
+        published_to_date: searchParams.get('published_to_date') || '',
     });
 
-    const revalidator = useRevalidator();
-
     const { roles, assignedEngagements } = useAppSelector((state) => state.user);
 
     const canViewPrivateEngagements = roles.includes(USER_ROLES.VIEW_PRIVATE_ENGAGEMENTS);
 
     const canViewAllCommentStatus = roles.includes(USER_ROLES.SHOW_ALL_COMMENT_STATUS);
 
     useEffect(() => {
-        updateURL(searchFilter, paginationOptions, searchOptions);
-        revalidator.revalidate();
+        {
+            const searchData = {
+                page: paginationOptions.page.toString(),
+                size: paginationOptions.size.toString(),
+                sort_key: paginationOptions.nested_sort_key?.toString(),
+                sort_order: paginationOptions.sort_order as 'asc' | 'desc' | undefined,
+                search_text: searchFilter.value,
+                engagement_status: searchOptions.status_list.map((status) => status.toString()),
+                created_from_date: searchOptions.created_from_date || undefined,
+                created_to_date: searchOptions.created_to_date || undefined,
+                published_from_date: searchOptions.published_from_date || undefined,
+                published_to_date: searchOptions.published_to_date || undefined,
+            };
+            // Filter out properties with empty strings or undefined values
```

Comment:
This is a nice little piece of code! Are any of the search params required or are all of them optional?

Reply:
They're optional! Making a request with no parameters uses the defaults on the API side